internal/storage/storage.go

```diff
@@ -11,4 +12,7 @@ type Storage interface {
 	Load(ctx context.Context, owner client.Object) (fs.FS, error)
 	Store(ctx context.Context, owner client.Object, bundle fs.FS) error
 	Delete(ctx context.Context, owner client.Object) error
+
+	http.Handler
+	URLFor(ctx context.Context, owner client.Object) (string, error)
```

Comment:
Perhaps debatable, but my thought process here was:
1. Every bundle provisioner MUST populate `contentURL`
2. If a provisioner uses a storage implementation to store/serve bundles, it needs a way to ask that storage implementation where it's serving a bundle so that it can satisfy 1.
3. Only the storage implementation knows the URL for a particular bundle because it is the thing serving that bundle.

Ergo: put both of these methods in the core storage interface.